datafusion/common/src/scalar.rs

```diff
@@ -5185,6 +5190,25 @@ mod tests {
         assert_eq!(1, arr.len());
     }
 
+    #[test]
+    fn test_newlist_timestamp_zone() {
+        let s: &'static str = "UTC";
+        let values = vec![ScalarValue::TimestampMillisecond(Some(1), Some(s.into()))];
+        let arr = ScalarValue::new_list(
+            &values,
```

Comment:
Does this fail if the timezone provided by one of the values is different than the one provided by the datatype argument?

Reply:
Yes, the test fails (and test as is fails on master).